src/main/java/io/proximax/sdk/infrastructure/Http.java

```diff
@@ -26,36 +28,58 @@
  * base HTTP repository implementation, keeping track of the API, HTTP client and mapper
  */
 public class Http {
-    protected final BlockchainApi api;
-    protected final HttpClient client;
-    protected Gson gson;
-    
-    /**
-     * create and initialize new instance for specified API
-     * 
-     * @param api the main API
-     */
-    Http(BlockchainApi api) {
-        this.api = api;
-        this.client = new OkHttpHttpClient(api);
-        // gson instance
-        this.gson = new Gson();
-    }
-
-    /**
-     * throw RuntimeException on error or return body of the response
-     * 
-     * @param response response to examine
-     * @return body of the response as string
-     */
-    static String mapStringOrError(final HttpResponse response) {
-       if (response.getCode() < 200 || response.getCode() > 299) {
-           throw new RuntimeException(response.getStatusMessage());
-       }
-       try {
-           return response.getBodyString();
-       } catch (IOException e) {
-           throw new RuntimeException(e.getMessage());
-       }
+   protected final BlockchainApi api;
+   protected final HttpClient client;
+   protected final Gson gson;
+
+   /**
+    * create and initialize new instance for specified API
+    * 
+    * @param api the main API
+    */
+   Http(BlockchainApi api) {
+      Validate.notNull(api, "api has to be provided");
+      this.api = api;
+      this.client = new OkHttpHttpClient(api);
+      // gson instance
+      this.gson = new Gson();
+   }
+
+   /**
+    * @return the api
+    */
+   public BlockchainApi getApi() {
+      return api;
+   }
+
+   /**
+    * @return the client
+    */
+   public HttpClient getClient() {
+      return client;
+   }
+
+   /**
+    * @return the gson
+    */
+   public Gson getGson() {
+      return gson;
+   }
+
+   /**
+    * throw RuntimeException on error or return body of the response
+    * 
+    * @param response response to examine
+    * @return body of the response as string
+    */
+   static String mapStringOrError(final HttpResponse response) {
+      if (response.getCode() < 200 || response.getCode() > 299) {
+         throw new RuntimeException(response.getStatusMessage());
```

Comment:
minor: If the response is "informational" (1xx) then it will still throw a RuntimeException.

Reply:
This is something that was there from original nemtech implementation. We are not expecting redirects or informational responses as those are not handled and would end by unexpected error down the pipeline (missing response body etc). It is definitely something to consider though and I raised issue #108 for that

Reply:
sorry for reformatting some of the classes. you can disable the whitespace changes from the diff (cog on the top of the screen) to get cleaner picture of the changes in respective classes